src/io/flutter/vmService/HeapMonitor.java

```diff
@@ -134,40 +128,48 @@ private void collectMemoryUsage() {
       return;
     }
 
-    // Stash count so we can know when we've processed them all.
-    final int isolateCount = isolateRefs.size();
-
     final List<MemoryUsage> memoryUsage = new ArrayList<>();
 
+    final CountDownLatch latch = new CountDownLatch(isolateRefs.size());
+
     for (IsolateRef isolateRef : isolateRefs) {
       vmServiceWrapper.getVmService().getMemoryUsage(isolateRef.getId(), new GetMemoryUsageConsumer() {
-        private int errorCount = 0;
-
         @Override
         public void received(MemoryUsage usage) {
           memoryUsage.add(usage);
-
-          // Only update when we're done collecting from all isolates.
-          if ((memoryUsage.size() + errorCount) == isolateCount) {
-            notifyListeners(memoryUsage);
-          }
+          latch.countDown();
         }
 
         @Override
         public void received(Sentinel sentinel) {
-          errorCount++;
+          latch.countDown();
         }
 
         @Override
         public void onError(RPCError error) {
-          errorCount++;
+          // {"jsonrpc":"2.0","error":{"code":-32603,"message":"UnimplementedError..."}}
```

Comment:
@grouma @jakemac53 @alan-knight - I'm using the return error code from the call to the vm service's `getMemoryUsage()` API to detect whether the current connection supports that call. Is `-32603` a good code to use for that? Is it a 'I don't support that API' code? I note that the message I get is essentially a stack trace, so I don't know if the code is 'unsupported' or 'the server threw an exception'.

```
==> {"jsonrpc":"2.0","id":"41","method":"getMemoryUsage","params":{"isolateId":"1"}}
```

```
<== {"jsonrpc":"2.0","error":{"code":-32603,"message":"UnimplementedError
	#0      ChromeProxyService.getMemoryUsage (package:dwds/src/services/chrome_proxy_service.dart:309:5)\n<asynchronous suspension>\n
	#1      VmServerConnection._delegateRequest (package:vm_service/vm_service.dart:1017:51)\n<asynchronous suspension>\n
	#2      _rootRunUnary (dart:async/zone.dart:1132:38)\n
	#3      _CustomZone.runUnary (dart:async/zone.dart:1029:19)\n
	#4      _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)\n
```

Reply:
It looks like we should really be returning [-32601](https://www.jsonrpc.org/specification#error_object) - I would check for that code and we should change our handling to return that for unimplemented errors

Reply:
https://github.com/dart-lang/webdev/issues/781

Reply:
Makes sense. I'll handle both for now, and file an issue on dwds to move to returning -32601:Method not found.

Reply:
ah, filed already as https://github.com/dart-lang/webdev/issues/781